apps/src/lib/wallet/derivation_path.rs

```diff
@@ -0,0 +1,167 @@
+use core::fmt;
+use std::str::FromStr;
+
+use derivation_path::{ChildIndex, DerivationPath as DerivationPathInner};
+use namada::types::key::SchemeType;
+use thiserror::Error;
+use tiny_hderive::bip44::{
+    DerivationPath as HDeriveDerivationPath,
+    IntoDerivationPath as IntoHDeriveDerivationPath,
+};
+use tiny_hderive::Error as HDeriveError;
+
+const ETH_COIN_TYPE: u32 = 60;
+const NAMADA_COIN_TYPE: u32 = 877;
+
+#[derive(Error, Debug)]
+pub enum DerivationPathError {
+    #[error("invalid derivation path: {0}")]
+    InvalidDerivationPath(String),
+}
+
+#[derive(Clone)]
+pub struct DerivationPath(DerivationPathInner);
+
+impl DerivationPath {
+    fn new<P>(path: P) -> Self
+    where
+        P: Into<Box<[ChildIndex]>>,
+    {
+        Self(DerivationPathInner::new(path))
+    }
+
+    pub fn is_compatible(&self, scheme: SchemeType) -> bool {
+        let mut it = self.0.into_iter();
+        let _ = it.next();
+        if let Some(coin_type) = it.next() {
+            let coin_type = coin_type.to_u32();
+            match scheme {
+                SchemeType::Ed25519 => coin_type == NAMADA_COIN_TYPE,
+                SchemeType::Secp256k1 => coin_type == ETH_COIN_TYPE,
+                _ => true,
+            }
+        } else {
+            true
+        }
+    }
+
+    fn bip44_base_indexes_for_scheme(scheme: SchemeType) -> Vec<ChildIndex> {
+        vec![
+            ChildIndex::Hardened(44),
+            match scheme {
+                SchemeType::Secp256k1 => ChildIndex::Hardened(ETH_COIN_TYPE),
+                SchemeType::Ed25519 => ChildIndex::Hardened(NAMADA_COIN_TYPE),
+                SchemeType::Common => unimplemented!("not implemented"),
+            },
+        ]
+    }
+
+    fn bip44(
+        scheme: SchemeType,
+        account: Option<u32>,
+        change: Option<u32>,
+        address: Option<u32>,
```

Comment:
This is only ever used with all 0s. Will we ever need to call these with None?

Reply:
I will remove `Option` from the parameters.